pkgs/_macro_host/lib/src/package_config.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:dart_model/dart_model.dart';
+import 'package:package_config/package_config.dart' as package_config;
+
+/// Reads a package config to determine information about macros.
+class PackageConfig {
+  final Uri uri;
```

Comment:
This should probably just take in a PackageConfig instance - and have a factory constructor or static method  which can construct that from a URI. That will make it more testable and avoid re-reading/parsing it on every call to `lookupMacroImplementation`.

Reply:
Done.